libvast/src/format/syslog.cpp

```diff
@@ -12,9 +12,148 @@
  ******************************************************************************/
 
 #include "vast/format/syslog.hpp"
+#include "vast/error.hpp"
 
-namespace vast::format::mrt {
+namespace vast {
+namespace format {
+namespace syslog {
 
-// TODO: implement functions here that do not belong into the header file. 
 
-} // namespace vast::format::mrt
+caf::error reader::schema(vast::schema x) {
+  auto xs = {
+    &types_.header_type,
+    &types_.structured_data_type,
+  };
+  return replace_if_congruent(xs, x);
+}
+
+vast::schema reader::schema() const {
+  vast::schema sch;
+  sch.add(types_.header_type);
+  sch.add(types_.structured_data_type);
+  return sch;
+}
+
+reader::reader(caf::atom_value table_slice_type,
+               const caf::settings& /*options*/,
+               std::unique_ptr<std::istream> in)
+  : super(table_slice_type) {
+  // Syslog header type.
+  // 
+  // TODO could it be better to have one record_type which describes a syslog message
```

Comment:
Since header:message is a 1:1 mapping that makes sense. In that case, you have a single record type, a single builder, and a single parser.